src/error.rs

```diff
@@ -42,3 +56,58 @@ impl From<str::Utf8Error> for MacaroonError {
         MacaroonError::DeserializationError(format!("{}", error))
     }
 }
+
+impl std::error::Error for MacaroonError {
+    // Note: `description()` is considered deprecated in standard library API; the Display
+    // implementation below is more important
+    fn description(&self) -> &str {
+        match *self {
+            MacaroonError::InitializationError => "initialization error",
+            MacaroonError::CryptoError(_) => "cryptography error",
+            MacaroonError::IncompleteMacaroon(_) => "incomplete macaroon",
+            MacaroonError::IncompleteCaveat(_) => "incomplete caveat",
+            MacaroonError::DeserializationError(_) => "error deserializing macaroon",
+            MacaroonError::CaveatNotSatisfied(_) => "one or more macaroon caveats not satisfied",
+            MacaroonError::DischargeNotUsed => "one or more macaroon discharged not used",
+            MacaroonError::InvalidSignature => "macaroon signature not correct",
+        }
+    }
+}
```

Comment:
Thanks for also adding this `impl`, but since our MSRV is quite a bit higher than 1.42, should we really bother adding this to just be removed later?